java/code/src/com/redhat/rhn/domain/image/ImageInfoFactory.java

```diff
@@ -443,13 +530,119 @@ public static Checksum convertChecksum(
     }
 
     /**
-     * @return the image build history
+     * @return the image repo digests
      */
-    public static List<ImageBuildHistory> listBuildHistory() {
-        CriteriaQuery<ImageBuildHistory> criteria = getSession()
+    public static List<ImageRepoDigest> listImageRepoDigests() {
+        CriteriaQuery<ImageRepoDigest> criteria = getSession()
                 .getCriteriaBuilder()
-                .createQuery(ImageBuildHistory.class);
-        criteria.from(ImageBuildHistory.class);
+                .createQuery(ImageRepoDigest.class);
+        criteria.from(ImageRepoDigest.class);
         return getSession().createQuery(criteria).getResultList();
     }
+
+    /**
+     * Set image revision to a number higher than other images with the same
+     * name and version
+     * @param image the image info.
+     */
+    public static void updateRevision(ImageInfo image) {
+        CriteriaBuilder builder = getSession().getCriteriaBuilder();
+        CriteriaQuery<ImageInfo> query = builder.createQuery(ImageInfo.class);
+        Root<ImageInfo> root = query.from(ImageInfo.class);
+
+        query.where(builder.and(builder.equal(root.get("name"), image.getName()),
+                                builder.equal(root.get("version"), image.getVersion())))
+                .orderBy(builder.desc(root.get("revisionNumber")));
```

Comment:
the store is not important in here?

Reply:
Having different images in different stores with the same name-version-revision is quite confusing, IMHO.
Not including store in this query ensures that name-version-revision is globally unique.